In [1]:
import json
import os
import pandas as pd

In [2]:
# Path to corpus file
multihop_corpus_path = os.path.join("..", "data", "Multi-hop_RAG_dataset", "corpus.json")

# Read JSON
with open(multihop_corpus_path, "r", encoding="utf-8") as f:
    corpus = json.load(f)

In [3]:
# Convert corpus data into df
corpus_as_df = pd.DataFrame(corpus)
# Get descriptive stats
corpus_as_df.describe()

title         author  \
count                                                 609            545   
unique                                                609            300   
top     Meta to offer ad-free subscription in Europe i...  Natasha Lomas   
freq                                                    1             14   

               source               published_at category  \
count             609                        609      609   
unique             49                        606        6   
top     Sporting News  2023-11-17T21:47:19+00:00   sports   
freq              101                          2      211   

                                                      url  \
count                                                 609   
unique                                                609   
top     https://techcrunch.com/2023/10/30/meta-ad-free...   
freq                                                    1   

                                                     body  
count                                                 609  
unique                                                609  
top     Meta is to offer an ad-free subscription versi...  
freq                                                    1

This corpus comprises 609 articles from 300 different authors, 49 different sources and 6 different categories.

In [ ]:
# Detail on authors
corpus_as_df["author"].unique()

array([None, 'Stan Choe', 'Colum Dell, Yardbarker', 'Dan Treacy',
       'Ian Valentino', 'Rob Miech', 'Arunima Bhanot',
       'The New York Times', 'Morgan Little', 'Jacquelyn Melinek',
       'Kyle Wiggers', 'Robert Calcutt', 'Will Unwin', 'David Pierce',
       'Rebecca Bellan', 'Jeffrey Parkin', 'Johnny Yu',
       'Laura Hampson and Meredith Clark', 'Dilvin Yasa', 'Amber Raiken',
       'Christy Doran', 'Catherine Shu', 'Natasha Lomas', 'Sarah Perez',
       'Cade Metz, Karen Weise, Nico Grant and Mike Isaac',
       'Ingrid Lunden', 'Ben Weiss', 'Tim Biggs', 'David Suggs',
       'Bryan Murphy', 'Kyle Irving', 'Romain Dillet', 'Vinnie Iyer',
       'Kyle Bonn', 'Haje Jan Kamps', 'Nell Geraets', 'Chelsea Ritschel',
       'Alyssa Stringer', 'Jessica Conditt', 'Laura Snapes',
       'Elizabeth Lopatto', 'Paul Sawers', 'Pete Volk',
       'Nick Brinkerhoff', 'Dave Richard', 'Shanna McCarriston',
       'Martin Derbyshire', 'Sam Eggleston', 'James Wallace', 'Rob Smyth',
       '', '

In [ ]:
# Detail on sources
corpus_as_df["source"].unique()

array(['Mashable', 'The Sydney Morning Herald',
       'Cnbc | World Business News Leader', 'Yardbarker', 'Sporting News',
       'Essentially Sports', 'The New York Times', 'TechCrunch',
       'TalkSport', 'The Guardian', 'The Verge', 'Polygon',
       'The Independent - Life and Style', 'FOX News - Entertainment',
       'The Age', 'The Roar | Sports Writers Blog', 'Fortune', 'Engadget',
       'CBSSports.com', 'FOX News - Health', 'FOX News - Lifestyle',
       'Globes English | Israel Business Arena', 'Hacker News',
       'Eos: Earth And Space Science News', 'Music Business Worldwide',
       'Sky Sports', 'Business Line',
       'Scitechdaily | Science Space And Technology News 2017',
       'Zee Business',
       'Business Today | Latest Stock Market And Economy News India',
       'Sport Grill', 'Wide World Of Sports', 'Financial Times',
       'Iot Business News', 'BBC News - Technology',
       'The Independent - Sports', 'Yahoo News',
       'Live Science: The Most Interest

In [ ]:
# Detail on category
corpus_as_df["category"].unique()

array(['entertainment', 'business', 'sports', 'technology', 'health',
       'science'], dtype=object)

In [ ]:
# Detail on article's wordcount

# Add new feature with word count of each article
corpus_as_df["body_wordcount"] = corpus_as_df["body"].str.split().str.len()
# Get descriptive stats
corpus_as_df["body_wordcount"].describe()



count      609.000000
mean      1746.008210
std       1338.843744
min        839.000000
25%       1036.000000
50%       1298.000000
75%       1938.000000
max      12387.000000
Name: body_wordcount, dtype: float64

The median wordcount is ~1300 words, roughly equivalent to between 3.5 and 5 pages depending on the particular format.